dandiapi/api/views/asset.py

```diff
@@ -88,12 +89,18 @@ def raise_if_unauthorized(self):
         asset_id = self.kwargs.get('asset_id')
         if asset_id is not None:
             asset = get_object_or_404(Asset, asset_id=asset_id)
-            if asset.embargoed_blob is not None:
+
+            # TODO: When EmbargoedZarrArchive is implemented, check that as well
+            if asset.blob and asset.blob.embargoed:
                 if not self.request.user.is_authenticated:
                     # Clients must be authenticated to access it
                     raise NotAuthenticated
-                if not self.request.user.has_perm('owner', asset.embargoed_blob.dandiset):
-                    # The user does not have ownership permission
+
+                # User must be an owner on any of the dandisets this asset belongs to
+                asset_dandisets = Dandiset.objects.filter(versions__in=asset.versions.all())
+                user_dandisets = get_objects_for_user(self.request.user, 'owner', Dandiset)
+                intersection = asset_dandisets & user_dandisets
+                if not intersection.exists():
                     raise PermissionDenied
```

Comment:
@jjnesbitt I opened a PR on top of this one to move some of this Python logic into SQL + add some joins to decrease the query overhead here - #1912